### PR TITLE
Feature/pyspark2.4.0 compatible

### PR DESCRIPTION
**Pull request by qfliu**

### CODE REVIEW

**Comment by qfliu** [comment]
To ensure the code works with both pyspark2.2.0 and pyspark2.4.0, created two dockerfile to test different versions of pyspark. Not sure if this is the best approach in this case.

**Comment by skliarpawlo** [comment]
I think we don't want to support multiple versions of pyspark in a long run in master. That's why I'd propose to tag/mark last version supporting pyspark2.2 - so we can branch from it for hotfixes, and keep master with pyspark2.4 support only

File: tests/integration/test_writer.py [hunk 1]
```diff
@@ -101,6 +101,7 @@ def test_write_cassandra(self):
             table='test_writer',
             consistency='ONE',
             mode='overwrite',
+            options={'confirm.truncate': True},
```
Comment (skliarpawlo) [inline tests/integration/test_writer.py]: let's comment why is this needed. Probably also add this to c* writer documentation - as it's also our public thing

**Comment by qfliu** [comment]
> I think we don't want to support multiple versions of pyspark in a long run in master. That's why I'd propose to tag/mark last version supporting pyspark2.2 - so we can branch from it for hotfixes, and keep master with pyspark2.4 support only. Thoughts?

**Comment by qfliu** [comment]
> I think we don't want to support multiple versions of pyspark in a long run in master. That's why I'd propose to tag/mark last version supporting pyspark2.2 - so we can branch from it for hotfixes, and keep master with pyspark2.4 support only. Thoughts?

I am ok with only supporting 2.4. Though it would be nice if the code can work in multiple versions, which is still the case right now. I will remove the DockerFile for pyspark2.2.0.

File: README.rst [hunk 2]
```diff
@@ -75,11 +75,11 @@ Supported Spark Versions
 
 At the moment we support:
 
-+-------------------------------------------+
-| sparkly 2.x | Spark 2.0.x and Spark 2.1.x |
-+-------------------------------------------+
-| sparkly 1.x | Spark 1.6.x                 |
-+-------------------------------------------+
++---------------------------------------------------------------------------+
+| sparkly 2.x | Spark 2.0.x and Spark 2.1.x and Spark 2.2.x and Spark 2.4.x |
```
Comment (skliarpawlo) [inline README.rst]: if we are not sure that spark 2.4.x and spark 2.2/1.x are compatible, let's have a separate row here like sparkly >= 2.7 is spark 2.4 compatible

File: tests/integration/test_writer.py [hunk 3]
```diff
@@ -101,6 +101,10 @@ def test_write_cassandra(self):
             table='test_writer',
             consistency='ONE',
             mode='overwrite',
+            # overwrite would first perform truncation.
```
Comment (skliarpawlo) [inline tests/integration/test_writer.py]: [optional] Would it be helful to add this to documentation of cassandra writer?
